tests/gdb-tests/tests/test_command_killthreads.py

```diff
@@ -1,21 +1,37 @@
+import time
+
 import gdb
 
 import tests
 
 REFERENCE_BINARY_THREADS = tests.binaries.get("multiple_threads.out")
 
 
+def wait_until(predicate: callable, timeout: int = 5):
+    """
+    Waits until the predicate returns True or timeout is reached.
+    """
+    counter = 0
+    while True:
+        if predicate():
+            return True
+        time.sleep(0.1)
+        counter += 0.1
+        if counter > timeout:
+            assert False, "Timeout reached"
+
+
 def test_command_killthreads_kills_all_threads_except_current(start_binary):
     start_binary(REFERENCE_BINARY_THREADS)
 
     gdb.execute("break break_here")
     gdb.execute("run")
-    assert len(gdb.selected_inferior().threads()) != 1
+    wait_until(lambda: len(gdb.selected_inferior().threads()) != 1)
```

Comment:
We should probably just change the multiple_threads example so that the break_here is triggered only after other thread[s] are spawned. In other words, the other threads should synchronize with the main thread so that break_here is only called after we are in the desired state 😅

Reply:
I've added the semaphores, but it still fails, but later. I think there is a delay between when the thread dies and gdb reports it as gone